lang/talon/talon.py

```diff
@@ -73,7 +70,57 @@ def on_ready():
 app.register("ready", on_ready)
 
 
-@ctx.action_class("user")
+@mod.action_class
+class Actions:
+    def talon_code_insert_function(text: str, selection: str):
+        """inserts talon-specific action call"""
+
+    def talon_code_enable_tag(tag: str):
+        """enables tag in either python or talon files"""
+
+    def talon_code_enable_setting(setting: str):
+        """asserts setting in either python or talon files"""
+
+
+@ctx_talon.action_class("user")
+class TalonActions:
+    def talon_code_insert_function(text: str, selection: str):
+        text += f"({selection or ''})"
+        actions.user.paste(text)
+        actions.edit.left()
+
+    def talon_code_enable_tag(tag: str):
+        """enables tag in either python or talon files"""
+        actions.user.paste(f"tag(): {tag}")
+
+    def talon_code_enable_setting(setting: str):
+        """asserts setting in either python or talon files"""
+        actions.user.paste(f"{setting} = ")
+
+
+@ctx_talon_python.action_class("user")
+class TalonPythonActions:
+    def talon_code_insert_function(text: str, selection: str):
+        text = f"actions.{text}({selection or ''})"
+        actions.user.paste(text)
+        actions.edit.left()
+
+    def talon_code_enable_tag(tag: str):
+        """enables tag in either python or talon files"""
+        actions.user.paste(f'ctx.tags = ["{tag}"]')
+        if not tag:
+            actions.edit.left()
+            actions.edit.left()
+
+    def talon_code_enable_setting(setting: str):
+        """asserts setting in either python or talon files"""
+        actions.user.paste(f'ctx.settings["{setting}"] = ')
+        if not setting:
+            actions.edit.left()
+            actions.edit.left()
```

Comment:
This does not move the cursor far enough left. Consider using `actions.user.insert_between` instead.